### DISCUSSION TITLE
Custom label for dashboard menu section

### DISCUSSION THREAD

**Discussion by ajitbohra** [laravel/nova-issues#3944] [Ideas & Feature Requests]
- Laravel Version: 9.2
- Nova Version: 4.1
- PHP Version: 8.1.4

### Description:

Nova Docs: MenuSection for Dashboard 

``` php
use App\Nova\Dashboards\Sales;
use Laravel\Nova\Menu\MenuSection;

MenuSection::dashboard('Sales', Sales::class),
```

This throws an error as `MenuSection::dashboard` accepts only 1 parameter which is dashboard class

So another way to set the menu text is to set `name` on the dashboard as mentioned in the docs but thats also not working
``` php
/**
 * Get the displayable name of the dashboard.
 *
 * @return string
 */
public function name()
{
    return 'User Insights';
}
```

From Nova Source: The menu section accepts only dashboard class and uses the label
``` php
  /**
     * Create a menu from dashboard class.
     *
     * @param  Dashboard  $dashboard
     * @return static
     */
    public static function dashboard($dashboard)
    {
        return with(new $dashboard(), function ($dashboard) {
            return static::make(
                $dashboard->label()
            )->path('/dashboards/'.$dashboard->uriKey());
        });
    }
```

So to set the menu label only thing that works is 
``` php
public function label()
{
    return 'Dashboard';
}

```

**Comment by ajitbohra** [comment]
Though this is labeled as `documentation` and we have the PR for the same. It would be good if menu text can be passed when registering the `menusection` as mentioned in the docs 

``` php
MenuSection::dashboard('Sales', Sales::class),
```

**Comment by crynobone** [comment]
You are suggesting a breaking change.

**Reply by ajitbohra** [reply]
In that case docs for `MenuSection` need update

**Comment by JnnisCanis** [comment]
Sorry guys for opening this old discussion again, but we are faced with a similar problem:
We want to have a different Headline on the page compared to the Menulink. But it looks like that even when using name and label with different values, it does not work.
This is our code


    public function name(): string
    {
        return 'Quartalsziele und aktuelle Zahlen für Q3 2023';
    }

    public function label(): string
    {
        return 'Quartalsziele';
    }

The menu-item and the page heading is just "Quartalsziele".
